bin/alluxio-mount.sh

```diff
@@ -95,39 +106,53 @@ function mount_ramfs_mac() {
   local num_sectors=$(${BIN}/alluxio runClass alluxio.util.HFSUtils ${MEM_SIZE} 512)
 
   # Format the RAM FS
-  echo "Formatting RamFS: ${TIER_PATH} ${num_sectors} sectors (${MEM_SIZE})."
+  echo "Formatting RamFS: ${1} ${num_sectors} sectors (${MEM_SIZE})."
   # Remove the "/Volumes/" part so we can get the name of the volume.
-  diskutil erasevolume HFS+ ${TIER_PATH/#\/Volumes\//} $(hdiutil attach -nomount ram://${num_sectors})
+  diskutil erasevolume HFS+ ${1/#\/Volumes\//} $(hdiutil attach -nomount ram://${num_sectors})
 }
 
 function umount_ramfs_mac() {
-  local device=$(df -l | grep ${TIER_PATH} | cut -d " " -f 1)
+  local device=$(df -l | grep $1 | cut -d " " -f 1)
   if [[ -n "${device}" ]]; then
-    echo "Unmounting ramfs at ${TIER_PATH}"
+    echo "Unmounting ramfs at ${1}"
     hdiutil detach -force ${device}
   else
-    echo "Ramfs is not currently mounted at ${TIER_PATH}"
+    echo "Ramfs is not currently mounted at ${1}"
   fi
 }
 
+function mount_ramfs_local_all() {
+for i in "${RAMDISKARRAY[@]}"
+do 
+  mount_ramfs_local $i
+done
+}
+
 function mount_ramfs_local() {
   if [[ $(uname -a) == Darwin* ]]; then
     # Assuming Mac OS X
-    umount_ramfs_mac
-    mount_ramfs_mac
+    umount_ramfs_mac $1
+    mount_ramfs_mac $1
   else
     # Assuming Linux
     check_space_linux
-    umount_ramfs_linux
-    mount_ramfs_linux
+    umount_ramfs_linux $1
+    mount_ramfs_linux $1
   fi
 }
 
+function umount_ramfs_local_all() {
+for i in "${RAMDISKARRAY[@]}"
+do 
+  umount_ramfs_local $i
+done
+}
+
 function umount_ramfs_local() {
   if [[ $(uname -a) == Darwin* ]]; then
-    umount_ramfs_mac
+    umount_ramfs_mac $1
```

Comment:
same here

Reply:
^

Reply:
i think it is ok to leave this function parameterized right?

Reply:
okay, your call